src/kepler_model/train/extractor/extractor.py

```diff
@@ -160,6 +162,11 @@ def get_workload_feature_data(self, query_results, features):
             aggr_query_data = query_results[query].copy()
 
             if all(col in aggr_query_data.columns for col in container_id_cols):
+                if use_vm_metrics:
+                    aggr_query_data = aggr_query_data.loc[aggr_query_data["job"] == VM_JOB_NAME]
+                else:
+                    aggr_query_data = aggr_query_data.loc[aggr_query_data["job"] != VM_JOB_NAME]
```

Comment:
Don't think we need this else condition. To avoid unexpected job filtering on general case, I think we can filter only when enable use_vm_metrics.

Reply:
My concern is that if the filter is not added, job=vm metrics will also be pulled by the model server alongside baremetal metrics which is not a desirable result right?

Reply:
Just thinking that the label job is a common label that could be relabeled for example if we want to label the pod with the application job. However, I understand your points that we might want to get the metric of BM also for the case that we enable the VM. After some thought, I think we can keep it as you code.